src/screens/SearchScreen.js

```diff
@@ -0,0 +1,175 @@
+import React, {Component} from 'react';
+import {View, Text, ActivityIndicator, StyleSheet, Platform, Keyboard} from 'react-native';
+import {Button} from 'react-native-elements';
+import {MapView} from 'expo';
+import {Icon} from 'react-native-elements';
+
+import _ from 'lodash';
+
+import {CardSection, Card} from '../components/common/index';
+
+const SearchOptions = {
+    MAP: 'map',
+    LIST: 'list'
+};
+
+
+class SearchScreen extends Component {
+    state = {
+        region: {
+            longitude: -105.727939,
+            latitude: 39.695168,
+            longitudeDelta: 1,
+            latitudeDelta: 1
+        },
+        mapLoaded: false,
+        viewStyle: SearchOptions.MAP
+    };
+
+    toggleButton = () => {
+        const {viewStyle} = this.state;
+
+        const nextButtonName = viewStyle;
+        const newViewStyle = viewStyle === SearchOptions.MAP ? SearchOptions.LIST : SearchOptions.MAP;
+
+        this.props.navigation.setParams({buttonName: nextButtonName});
+        this.setState({viewStyle: newViewStyle});
+    };
+
+    componentDidMount() {
+        Keyboard.dismiss();
+
+        this.props.navigation.setParams({buttonName: SearchOptions.LIST, toggleButton: this.toggleButton});
+
+        this.setState({
+            mapLoaded: true
+        });
+    }
+
+    static renderLeftNavButton = ({buttonName, toggleButton}) => {
+        let preparedName = _.capitalize(buttonName);
+
+        if (Platform.OS === 'ios') {
+            return (
+                <Button
+                    title={preparedName}
+                    onPress={toggleButton}
+                    backgroundColor="rgba(0,0,0,0)"
+                    color="rgba(0,122,255,1)"
+                />
+            );
+        } else if (Platform.OS === 'android') {
+            // android-specific code for navigation here
+        }
+    };
```

Comment:
Above, the `renderLeftNavButton` is a `static` class method, so we can't do simple things like call `this`. Instead, we have to pass it a function from the instance to the `static` method and do it that way. Here, the `buttonName` property and `toggleButton` function are used to toggle the top-left of the navBar between `Map` and `List`.